src/evict/evict_inline.h

```diff
@@ -588,30 +588,44 @@ __wti_evict_hs_dirty(WT_SESSION_IMPL *session)
       ((uint64_t)(conn->evict->eviction_dirty_trigger * bytes_max) / 100));
 }
 
+/*
+ * __evict_check_user_ok_with_eviction --
+ *     Check if the user wants to abort eviction in this session. #private
+ */
+static bool
+__evict_check_user_ok_with_eviction(WT_SESSION_IMPL *session, bool busy)
+{
+    /* Ask the user only if eviction is optional and it's a user session. */
+    if (busy || F_ISSET(session, WT_SESSION_INTERNAL))
+        return (true);
+
+    WT_EVENT_HANDLER *event_handler = session->event_handler;
+    if (event_handler->handle_general != NULL &&
+      event_handler->handle_general(
+        event_handler, &S2C(session)->iface, &session->iface, WT_EVENT_EVICTION, NULL) != 0)
+        return (false);
+
+    return (true);
+}
+
 /* !!!
  * __wt_evict_app_assist_worker_check --
  *     Check if eviction trigger thresholds have reached to determine whether application threads
  *     should assist eviction worker threads with eviction of pages from the queues.
  *
  *     Input parameters:
- *       (1) `busy`: A flag indicating if the session is actively pinning resources, in which
- *            case dirty trigger is ignored.
+ *       (1) `busy`: A flag indicating if eviction is mandatory (true) or optional (false).
```

Comment:
After checking the code, I don't see that `busy` always guarantees eviction. But more of an indication whether we should consider specific parameters because we are pinning content, see `__wt_evict_needed`.

Reply:
I had the same concern @etienneptl. We can merge this and I will start a conversation in the storage about what the busy means, and create a new ticket if it deviates from the understood meaning in this PR. What do you think?

Reply:
Sure